nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs

```diff
@@ -182,7 +192,7 @@ await _apiHandler.IncrementMetric(new()
                     {
                         // TODO: this is inefficient, but not likely causing a bottleneck
                         var previousDependency = discoveredUpdatedDependencies.Dependencies
-                            .Single(d => d.Name == dependency.Name && d.Requirements.Single().File == Path.Join(discoveryResult.Path, project.FilePath).NormalizePathToUnix().EnsurePrefix("/"));
+                            .Single(d => d.Name == dependency.Name);
```

Comment:
Still need to investigate more on whether removing this check is okay or if there is a way to check for the correct file. Will be done before completing this PR.